src/utils/MSFraggerAdapter.cpp

```diff
@@ -678,45 +645,47 @@ class TOPPMSFraggerAdapter final :
     }
 
     QStringList process_params; // the actual process is Java, not MSFragger
-    process_params << "-Xmx" + QString::number(this->getIntOption_(TOPPMSFraggerAdapter::param_java_heapmemory)) + "m"
-        << "-jar" << this->executable.toQString()
-        << this->parameter_file_path.toQString();
-
-    for (const QString & input_file : this->input_files)
-    {
-      process_params << input_file;
-    }
+    process_params << "-Xmx" + QString::number(this->getIntOption_(java_heapmemory)) + "m"
+        << "-jar" << this->exe.toQString()
+        << this->parameter_file_path.toQString()
+        << link_name;
 
     QProcess process_msfragger;
     process_msfragger.setWorkingDirectory(this->working_directory);
 
     if (this->debug_level_ >= TOPPMSFraggerAdapter::LOG_LEVEL_VERBOSE)
     {
       writeDebug_("COMMAND LINE CALL IS:", 1);
-      String command_line = this->java_executable;
+      String command_line = this->java_exe;
       for (const String & process_param : process_params)
       {
         command_line += (" " + process_param);
       }
       writeDebug_(command_line, TOPPMSFraggerAdapter::LOG_LEVEL_VERBOSE);
     }
 
-    process_msfragger.start(this->java_executable.toQString(), process_params);
+    process_msfragger.start(this->java_exe.toQString(), process_params);
 
     if (process_msfragger.waitForFinished(-1) == false || process_msfragger.exitCode() != 0)
     {
       LOG_FATAL_ERROR << "FATAL: Invocation of MSFraggerAdapter has failed. Error code was: " << process_msfragger.exitCode() << std::endl;
       return EXTERNAL_PROGRAM_ERROR;
     }
 
-    // Copy the output files of MSFragger to the user location
-    for (int i = 0; i < this->input_files.size(); ++i)
+    // convert from pepXML to idXML
+    String pepxmlfile = File::removeExtension(link_name) + "." + "pepXML";
+    std::vector<PeptideIdentification> peptide_identifications;
+    std::vector<ProteinIdentification> protein_identifications;
+    PepXMLFile().load(pepxmlfile, protein_identifications, peptide_identifications);
```

Comment:
Should we overwrite the weird XTandem hack of MSFragger here?

Reply:
Actually I don't know if it is necessary (will probably be updated in the following MSFragger Version) - if you think it is important - I can change it.

Reply:
I think it would make more sense. In the end we have a lot of idxmls lying around with XTandem annotated. And for us it does not make sense as we do not directly use peptide prophet.